pgsanity/pgsanity.py

```diff
@@ -27,8 +48,12 @@ def check_file(filename=None, show_filename=False):
     else:
         with sys.stdin as filelike:
             sql_string = sys.stdin.read()
-
-    success, msg = check_string(sql_string)
+    # check for BOM-table and discard if present
+    nose = sql_string[0:len(BOM_UTF8)]
+    bom_present = check_for_bom(nose)
+    sql_string = sql_string[len(nose):] if bom_present else sql_string
+    success, msg = check_string(sql_string.decode("utf-8"))
+    # ^ The above call to decode() is safe for both ASCII and UTF-8 data.
```

Comment:
I had just noticed all of the other check\_*something*() -functions hanging out in this file and at the time I thought I'd go with the flow! I'm happy to rename the function and wrap up that logic all the way.